modules/core/src/main/scala/gem/math/Coordinates.scala

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.math
+
+import scala.math._
+import scalaz._, Scalaz._
+
+/** A point in the sky, given right ascension and declination. */
+final case class Coordinates(ra: RightAscension, dec: Declination) {
+
+  /**
+   * Offset these `Coordinates` by the given deltas, and indicate whether the declination crossed
+   * a pole; if so the right ascension will have been flipped 180°.
+   * @group Operations
+   */
+  def offsetWithCarry(dRA: HourAngle, dDec: Angle): (Coordinates, Boolean) =
+    dec.offset(dDec) match {
+      case (decʹ, false) => (Coordinates(ra.offset(dRA),      decʹ), false)
+      case (decʹ, true)  => (Coordinates(ra.flip.offset(dRA), decʹ), true)
+    }
+
+  /**
+   * Offset these `Coordinates` by the given deltas. If the declination crossed a pole the right
+   * ascension will have been flipped 180°.
+   * @group Operations
+   */
+  def offset(dRA: HourAngle, dDec: Angle): Coordinates =
+    offsetWithCarry(dRA, dDec)._1
+
+  /**
+   * Compute the offset between points, such that `a offset (a diff b) = b`.
+   * @group Operations
+   */
+  def diff(c: Coordinates): (HourAngle, Angle) =
+    (c.ra.toHourAngle - ra.toHourAngle, c.dec.toAngle - dec.toAngle)
+
+  /**
+   * Angular distance from `this` to `that`, always a positive angle in [0, 180]). Approximate.
+   * @see Algorithm at [[http://www.movable-type.co.uk/scripts/latlong.html Movable Type Scripts]].
+   */
+  def angularDistance(that: Coordinates): Angle = {
+    val φ1 = this.dec.toAngle.toDoubleRadians
+    val φ2 = that.dec.toAngle.toDoubleRadians
+    val Δφ = (that.dec.toAngle - this.dec.toAngle).toDoubleRadians
+    val Δλ = (that.ra.toAngle  - this.ra.toAngle) .toDoubleRadians
+    val a  = sin(Δφ / 2) * sin(Δφ / 2) +
+             cos(φ1)     * cos(φ2)     *
+             sin(Δλ / 2) * sin(Δλ / 2)
+    Angle.fromDoubleRadians(2 * atan2(sqrt(a), sqrt(1 - a)))
+  }
+
+  /**
+   * Interpolate between `this` and `that` at a position specified by `f`, where `0.0` is `this`,
+   * `1.0` is `other`, and `0.5` is halfway along the great circle connecting them. Note that this
+   * computation is undefined where `f` is `NaN` or `Infinity`. Approximate.
+   * @see Algorithm at [[http://www.movable-type.co.uk/scripts/latlong.html Movable Type Scripts]].
+   */
+  def interpolate(that: Coordinates, f: Double): Coordinates = {
+    val δ = angularDistance(that).toDoubleRadians
+    if (δ === 0) this
+    else {
+      val φ1 = this.dec.toAngle.toDoubleRadians
+      val φ2 = that.dec.toAngle.toDoubleRadians
+      val λ1 = this.ra.toAngle.toDoubleRadians
+      val λ2 = that.ra.toAngle.toDoubleRadians
+      val a  = sin((1 - f) * δ) / sin(δ) // n.b. this line is wrong on the web page
```

Comment:
Well, same at 0° … the math works out since `Double` never overflows and `atan2` understands infinity I guess. I haven't really thought about how this works. 

```scala
scala> import gem.math.Coordinates._
import gem.math.Coordinates._
 
scala> Zero.angularDistance(Zero).toDoubleRadians
res2: Double = 0.0
 
scala> Zero.interpolate(Zero, 0)
res3: gem.math.Coordinates = Coordinates(RA(0:00:00.000000),Dec(0:00:00.000000))
 
scala> SouthPole.angularDistance(NorthPole).toDoubleRadians
res4: Double = 3.141592653589793
 
scala> SouthPole.interpolate(NorthPole, 0.5)
res5: gem.math.Coordinates = Coordinates(RA(12:00:00.000000),Dec(0:00:00.000000))
```